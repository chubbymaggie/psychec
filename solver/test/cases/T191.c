Result generate(double val)
{
    List list;            /* No declaration of `List' available. */
    list = 0;             /* `List' is a pointer or an integral. */
    list = make();        /* Function `make' must return a `List'. */
    list->data = val;     /* List has a member of type `double'. */
    list->next = list;    /* And it's a recursive type. */
    if (analyse(list))    /* Function `analyse' must take a `List'. */
        return "OK";      /* We got `generate's return type. */
    return FAIL_MACRO(1); /* Unexpanded macro, but syntactically valid. */
}
