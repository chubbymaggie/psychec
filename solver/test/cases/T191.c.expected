/* Forward declarations */
typedef  struct TYPE_4__   TYPE_1__ ;

/* Type definitions */
struct TYPE_4__ {double data; struct TYPE_4__* next; } ;
typedef  char* Result ;
typedef  TYPE_1__* List ;

/* Variables and functions */
 char* FAIL_MACRO (int) ; 
 scalar_t__ analyse (TYPE_1__*) ;
 TYPE_1__* make () ; 
